crates/hir_def/src/lib.rs

```diff
@@ -281,11 +281,19 @@ impl_intern!(BlockId, BlockLoc, intern_block, lookup_intern_block);
 #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
 pub struct TypeParamId {
     pub parent: GenericDefId,
-    pub local_id: LocalTypeParamId,
+    pub local_id: LocalTypeOrConstParamId,
 }
 
 pub type LocalTypeParamId = Idx<generics::TypeParamData>;
```

Comment:
is this still used?

Reply:
My intent was to remove it, and it isn't used anywhere. Keeping all of them under a same type can help in saving some `.into()` and wrapper functions and using them in or patterns like this:
```rust
GenericParamId::TypeParamId(id) | GenericParamId::ConstParamId(id) => {
    id.parent.child_source(db).map(|source| match &source[id.local_id] {
        Either::Left(ast::TypeOrConstParam::Type(id)) => {
            ast::AnyHasAttrs::new(id.clone())
        }
        Either::Left(ast::TypeOrConstParam::Const(id)) => {
            ast::AnyHasAttrs::new(id.clone())
        }
        Either::Right(id) => ast::AnyHasAttrs::new(id.clone()),
    })
}
```

Since in most of the cases we want to do the same with `ConstParamId` and `TypeParamId`. But I don't have a strong opinion. If separating them is better, I can do it.

Reply:
We can do it like this, but then let's remove these types.